src/app/report/edit-display/encoding-button/encoding-button.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { ReportOrCheckpoint, ReportUtil } from '../../../shared/util/report-util';
+import { ToastService } from '../../../shared/services/toast.service';
+import { CustomEditorComponent } from '../../../custom-editor/custom-editor.component';
+
+@Component({
+  selector: 'app-encoding-button',
+  standalone: true,
+  imports: [],
+  templateUrl: './encoding-button.component.html',
+  styleUrl: './encoding-button.component.css',
+})
+export class EncodingButtonComponent implements OnChanges {
+  @Input({ required: true }) selectedNode!: ReportOrCheckpoint;
+  @Input({ required: true }) editor!: CustomEditorComponent;
```

Comment:
This makes the flow of data sharing between components very unlogical IMO, right now this flow looks a little like:
```mermaid
flowchart TD
    A[edit-display.component] --> B[encoding-button.component];
    A <--> C[custom-editor.component]
    B --> C
```
however it should be:
```mermaid
flowchart TD
    A[edit-display.component] <--> B[encoding-button.component];
    A <--> C[custom-editor.component]
```
I think it makes more sense to have all data sharing go from the parent to the child, so in this case it's best to have the encoding-button.component have an output so the parent can do with that output as it chooses, this way our components have a tight cohesion but are loosely coupled.

Reply:
I moved the this.editor.setNewReport(message) call to a method in the edit-display component which gets called on te output attribute eventemitter from the encoding-button component.